### DISCUSSION TITLE
Preserving Text Structure in Mixed-Language Receipt Images

### DISCUSSION THREAD

**Discussion by sifdinNh** [PaddlePaddle/PaddleOCR#13586] [Q&A]
Hello everyone,

I’m using PaddleOCR to extract structured data from receipt images, which can contain both Arabic and English text. Since PaddleOCR doesn't support recognizing multiple languages simultaneously, I’ve been using two instances—one for Arabic and one for English. I then compare the confidence levels of each instance to determine the correct language for each text segment, and this workaround has been effective.

However, I'm facing an issue with preserving the structure of the text lines as they appear on the receipts. The output from PaddleOCR does not maintain the original line structure, which is critical for my use case.

Is there a way to ensure that the text structure in the output matches the layout of the text on the receipt? Maintaining the correct structure is important for accurately extracting and interpreting the data.

Any advice or suggestions would be greatly appreciated.

**Comment by SWHL** [comment]
If you can get the box coordinates corresponding to the text boxes in these two languages, you can directly sort them by box to maintain the original order.

**Reply by sifdinNh** [reply]
Yes i can get the box coordinates for both languages
How can i make this happen

**Reply by SWHL** [reply]
Sort by upper left corner coordinates
